system-int-test/test/cmr/system_int_test/ingest/granule_ingest_test.clj

```diff
@@ -48,34 +56,53 @@
     (index/wait-until-indexed)
     ;; Make sure the granules reference the correct collection
     (is (= (:concept-id new-coll)
-           (get-in (mdb/get-concept (:concept-id gran1) (:revision-id gran1))
-                   [:extra-fields :parent-collection-id])))
+           (get-granule-parent-collection-id gran1)))
 
     (is (= (:concept-id new-coll)
-           (get-in (mdb/get-concept (:concept-id gran2) (:revision-id gran2))
-                   [:extra-fields :parent-collection-id])))
+           (get-granule-parent-collection-id gran2)))
     (is (= (:concept-id new-coll)
-           (get-in (mdb/get-concept (:concept-id gran3) (:revision-id gran3))
-                   [:extra-fields :parent-collection-id])))))
+           (get-granule-parent-collection-id gran3)))))
 
 (deftest granule-change-parent-collection-test
   (testing "Cannot change granule's parent collection"
-    (let [coll1 (d/ingest-umm-spec-collection "PROV1" (data-umm-c/collection {:EntryTitle "coll1"
-                                                                              :ShortName "short1"
-                                                                              :Version "V1"
-                                                                              :native-id "native1"}))
-          coll2 (d/ingest-umm-spec-collection "PROV1" (data-umm-c/collection {:EntryTitle "coll2"
-                                                                              :ShortName "short2"
-                                                                              :Version "V2"
-                                                                              :native-id "native2"}))
-          _ (d/ingest "PROV1" (dg/granule-with-umm-spec-collection coll1 (:concept-id coll1) {:granule-ur "gran1"}))
-          {:keys [status errors]} (d/ingest "PROV1"
-                                            (dg/granule-with-umm-spec-collection coll2 (:concept-id coll2) {:granule-ur "gran1"})
-                                            {:allow-failure? true})]
+    (let [coll1 (d/ingest-umm-spec-collection
+                 "PROV1" (data-umm-c/collection {:EntryTitle "coll1"
+                                                 :ShortName "short1"
+                                                 :Version "V1"
+                                                 :native-id "native1"}))
+          coll2 (d/ingest-umm-spec-collection
+                 "PROV1" (data-umm-c/collection {:EntryTitle "coll2"
+                                                 :ShortName "short2"
+                                                 :Version "V2"
+                                                 :native-id "native2"}))
+          gran1 (d/item->concept
+                 (dg/granule-with-umm-spec-collection
+                  coll1
+                  (:concept-id coll1)
+                  {:granule-ur "gran1"
```

Comment:
is specifying the granule-ur necessary?

Reply:
when I change gran2 ur to gran2 instead of gran1, the test passes.  When I remove the :granule-ur there is a unique one created by the granule-with-umm-spec-collection.  Either of these scenarios allows the test to pass.  Below is the relevant code in granule-with-umm-spec-collection:

```
         minimal-gran {:granule-ur (d/unique-str "ur")
                       :collection-ref coll-ref
                       ;; Including the parent collection concept id so it's available later.
                       :collection-concept-id collection-concept-id}
```

Reply:
Specifying granule-ur is unnecessary in this test. I am going to remove it. Thanks.